core/src/main/java/org/polypheny/db/catalog/Catalog.java

```diff
@@ -546,6 +559,14 @@ protected final boolean isValidIdentifier( final String str ) {
 
     public abstract List<CatalogColumnPlacement> getColumnPlacementsByColumn( long columnId );
 
+    public abstract ImmutableMap<Integer, ImmutableList<Long>> getColumnPlacementsByAdapter( long tableId );
+
+    public abstract ImmutableMap<Integer, ImmutableList<Long>> getPartitionPlacementsByAdapter( long tableId );
+
+    public abstract ImmutableMap<Integer, ImmutableList<Long>> getPartitionGroupsByAdapter( long tableId );
+
+    public abstract Long getPartitionGroupByPartition( long partitionId );
```

Comment:
Why returning a Long? Can this be null?

Reply:
No, that honestly was a lazy copy/paste error. Fixed it.